### PR TITLE
Write end_time_estimated in nexus file

### LINKED ISSUE #887: Write predicted end time in nexus file
We have seen more zocalo issues when we're putting the end time into the nexus file whilst zocalo is validating it. Of the form:
```
{'beamline': 'i03',
 'dc_comments': 'Hyperion: Xray centring - Diffraction grid scan of 30 by 13 '
                'images in 20.0 um by 20.0 um steps. Top left (px): [484,243], '
                'bottom right (px): [2206,989].',
 'file': '/dls/i03/data/2023/cm33866-4/xraycentring/TestProteinaseK/protk_4/protk_4_25_master.h5',
 'ispyb_expected_images': '390',
 'ispyb_wavelength': '0.976284',
 'reason': 'Unhandled KeyError exception reading '
           '/dls/i03/data/2023/cm33866-4/xraycentring/TestProteinaseK/protk_4/protk_4_25_master.h5'}
```
We can get round having to edit the nexus file after the collection if we instead put in the predicted time for the experiment. We can use the same time we use for the topup gate for this

### PR DESCRIPTION
**Pull request by noemifrisina**
Fixes #887 

Writes `end_time_estimated` field in the nexus file at the same time as `start_time` instead of `end_time`, removing the need to reopen the file at the end of the collection.
This is also more in line with the NXmx standard.

### To test:
1. Check tests relating to the nexus write/handler/callback still make sense and pass

### CODE REVIEW

File: src/hyperion/external_interaction/nexus/write_nexus.py [hunk 1]
```diff
@@ -79,6 +77,9 @@ def create_nexus_file(self):
         initialised.
         """
         start_time = get_current_time()
+        est_end_time = get_predicted_end_time(
+            start_time, self.detector.exp_time * self.full_num_of_images
+        )
```
Comment (DominicOram) [inline src/hyperion/external_interaction/nexus/write_nexus.py:82]: Should: It's a bit of a faff to get current time, convert to string then convert back to get the end time. I think we could just do it in one function like `start_time, est_end_time = get_start_and_predicted_end_time(time_expected)`, which will be simpler

**Review by DominicOram — CHANGES_REQUESTED** [review]
This is great, thank you. Nice to remove so much code! One small comment in code.

File: src/hyperion/external_interaction/nexus/write_nexus.py (hunk 1 above)
Comment (noemifrisina) [reply src/hyperion/external_interaction/nexus/write_nexus.py]: Good point, done

File: src/hyperion/external_interaction/nexus/nexus_utils.py [hunk 2]
```diff
@@ -63,8 +63,12 @@ def create_goniometer_axes(
...
 
 
-def get_current_time():
-    return datetime.utcfromtimestamp(time.time()).strftime(r"%Y-%m-%dT%H:%M:%SZ")
+def get_start_and_predicted_end_time(time_expected: float) -> tuple[str, str]:
+    start = datetime.utcfromtimestamp(time.time())
+    end_est = start + timedelta(seconds=time_expected)
+    return start.strftime(r"%Y-%m-%dT%H:%M:%SZ"), end_est.strftime(
+        r"%Y-%m-%dT%H:%M:%SZ"
+    )
```
Comment (DominicOram) [inline src/hyperion/external_interaction/nexus/nexus_utils.py:71]: Nit: Slightly cleaner I think, take it or leave it.
```suggestion
def get_start_and_predicted_end_time(time_expected: float) -> tuple[str, str]:
    time_format = r"%Y-%m-%dT%H:%M:%SZ"
    start = datetime.utcfromtimestamp(time.time())
    end_est = start + timedelta(seconds=time_expected)
    return start.strftime(time_format), end_est.strftime(time_format)
```

**Review by DominicOram — APPROVED** [review]
Great, thank you! Minor suggestion in code, take it or leave it.

**Pull request merged**